tests/Amazon.CloudWatch.EMF.Tests/Model/DimensionSetTests.cs

```diff
@@ -25,6 +26,43 @@ public void AddDimension_30_Dimensions()
             Assert.Equal(dimensionSetSize, dimensionSet.DimensionKeys.Count);
         }
 
+        [Theory]
+        [InlineData(null, "value")]
+        [InlineData(" ", "value")]
+        [InlineData("ďïɱ", "value")]
+        [InlineData("dim", null)]
+        [InlineData("dim", " ")]
+        [InlineData("dim", "ⱱẵĺ")]
+        [InlineData(":dim", "val")]
+        public void AddDimension_WithInvalidValues_ThrowsInvalidDimensionException(string key, string value)
+        {
+            Assert.Throws<InvalidDimensionException>(() =>
+            {
+                var dimensionSet = new DimensionSet();
+                dimensionSet.AddDimension(key, value);
+            });
+        }
+
+        [Fact]
+        public void AddDimension_WithNameTooLong_ThrowsInvalidDimensionException()
+        {
+            Assert.Throws<InvalidDimensionException>(() =>
+            {
+                var dimensionSet = new DimensionSet();
+                dimensionSet.AddDimension(new string('a', Constants.MAX_DIMENSION_NAME_LENGTH + 1), "value");
```

Comment:
@markkuhn probably will be more readable if you move out the string construction into a separate variable like in another tests